docs(concepts): mention PEP 625 (#7189)
astral-sh/uv: docs/concepts/resolution.md

## Source distribution

> [PEP 625](https://peps.python.org/pep-0625/) specifies that packages must distribute source
> distributions as gzip tarball (`.tar.gz`) archives. As before this PEP, other formats were also
> allowed, uv supports reading and extracting archives that use the following formats:

Reviewer: ```suggestion
[PEP 625](https://peps.python.org/pep-0625/) specifies that packages must distribute source
distributions as gzip tarball (`.tar.gz`) archives. Prior to this specification, other archive formats
were allowed so uv also supports reading and extracting archives in the following formats:
```

Reviewer: Maybe something like this?

Author: Woops sorry, I force-pushed before your review, since it was still in draft (damn you're fast! 😄). Will incorporate your suggestions though, thanks!
